src/plots/linePlot.ts

```diff
@@ -425,6 +450,74 @@ export module Plots {
       return filteredDataIndices;
     }
 
+    private _filterDownsampling(dataset: Dataset, indices: number[]) {
+      let xAccessor = this.x().accessor;
+      let yAccessor = this.y().accessor;
+      let filteredIndices = this._filterDownsamplingSlope(dataset, indices, this.x().scale, this.y().scale, xAccessor, yAccessor);
+      return filteredIndices;
+    }
+
+    private _filterDownsamplingSlope(dataset: Dataset, indices: number[],
+      xScale: Scale<any, number>, yScale: Scale<any, number>, xAcessor: Accessor<any>, yAcessor: Accessor<any>) {
+      let data = dataset.data();
+      let filteredIndices: number[] = [];
+      let min: number;
+      let max: number;
+      let currentSlope: number;
+      if (indices.length === 0) {
+        return filteredIndices;
+      }
+      filteredIndices.push(indices[0]);
+      function belongToCurBucket(i: number) {
+        let p1x = xScale.scale(xAcessor(data[indices[i]], indices[i], dataset));
+        let p1y = yScale.scale(yAcessor(data[indices[i]], indices[i], dataset));
+        let p2x = xScale.scale(xAcessor(data[indices[i + 1]], indices[i + 1], dataset));
+        let p2y = yScale.scale(yAcessor(data[indices[i + 1]], indices[i + 1], dataset));
+        if (currentSlope == null) {
+          currentSlope = (Math.floor(p1x) === Math.floor(p2x)) ? Infinity : (p2y - p1y) / (p2x - p1x);
```

Comment:
I feel like checking `Math.abs(p2x - p1x)` is less a certain threshold is more accurate than comparing the floored values
but the difference is probably very minor